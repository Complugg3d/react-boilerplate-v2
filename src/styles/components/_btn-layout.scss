.btn-layout {
  color: white;
  background: $light-blue;
  border: none;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  padding: $s-size;
  text-decoration: none;
}
.btn-layout--link {
  background: none;
  border: none;
  color: white;
  padding: 0;
}

.btn-layout--secondary {
  background: $gray-steel;
}