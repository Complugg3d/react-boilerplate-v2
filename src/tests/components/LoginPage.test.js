import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { LoginPage } from '../../components/LoginPage';

test('should render LoginPage component correctly', () => {
  const wrapper = shallow(<LoginPage startLogin={() => { }} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

test('should call startLogin on btn click', () => {
  const startLogin = jest.fn();
  const wrapper = shallow(<LoginPage startLogin={startLogin} />);
  wrapper.find('button').prop('onClick')();
  expect(startLogin).toHaveBeenCalled()
});