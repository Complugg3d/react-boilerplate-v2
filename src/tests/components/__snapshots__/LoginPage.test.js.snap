// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage component correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Boilerlate
    </h1>
    <p>
      Tag line for app.
    </p>
    <button
      className="btn-layout"
      onClick={[Function]}
    >
      Login with Google
    </button>
  </div>
</div>
`;
